### PR TITLE
ft: add LifecycleTask functional tests expiration

### PR DESCRIPTION
**Pull request by philipyoo**
Only includes tests for non-versioned bucket expiration rule.

### CODE REVIEW

File: extensions/lifecycle/tasks/LifecycleTask.js [hunk 1]
```diff
@@ -667,6 +676,37 @@ class LifecycleTask extends BackbeatTask {
...
+    /**
+     * Only to be used when testing (when process.env.TEST_SWITCH).
+     * The idea is to adjust any "Days" or "NoncurrentDays" rules so that rules
+     * set with 1 day should expire, but any days set with 2+ days will not.
+     * Since Days/NoncurrentDays cannot be set to 0, this is a way to set the
+     * rule and test methods are working as intended.
+     * @param {object} rules - applicable rules
+     * @return {object} adjusted rules object
+     */
+    _adjustRules(rules) {
```
Comment (jonathan-gramain) [inline extensions/lifecycle/tasks/LifecycleTask.js]: Can you make this function name clearly stating it's for testing only?

File: extensions/lifecycle/tasks/LifecycleTask.js [hunk 2]
```diff
@@ -696,7 +736,8 @@ class LifecycleTask extends BackbeatTask {
         const daysSinceInitiated = this._findDaysSince(new Date(staleDate));
         const ncve = 'NoncurrentVersionExpiration';
         const ncd = 'NoncurrentDays';
-        const doesNCVExpirationRuleApply = (rules[ncve] && rules[ncve][ncd] &&
+        const doesNCVExpirationRuleApply = (rules[ncve] &&
```
Comment (jonathan-gramain) [inline extensions/lifecycle/tasks/LifecycleTask.js]: Maybe a better fix can be to change the test to `rules[ncve][ncd] !== undefined`? Then it would not require special handling of test case in the main code.

File: extensions/lifecycle/tasks/LifecycleTask.js [hunk 3]
```diff
@@ -751,7 +792,8 @@ class LifecycleTask extends BackbeatTask {
                 new Date(upload.Initiated));
             const abortRule = aRules.AbortIncompleteMultipartUpload;
             const doesAbortRuleApply = (abortRule &&
-                abortRule.DaysAfterInitiation &&
+                (abortRule.DaysAfterInitiation ||
```
Comment (jonathan-gramain) [inline extensions/lifecycle/tasks/LifecycleTask.js]: same here

File: package.json [hunk 4]
```diff
@@ -48,6 +48,7 @@
     "werelogs": "scality/werelogs#rel/7.4-beta"
   },
   "devDependencies": {
-    "mocha": "^3.3.0"
+    "mocha": "^3.3.0",
+    "s3": "scality/S3#rel/7.4-beta"
```
Comment (jonathan-gramain) [inline package.json]: The branch selection should not be part of the commit.

File: extensions/lifecycle/tasks/LifecycleTask.js (hunk 2 above)
Comment (philipyoo) [reply extensions/lifecycle/tasks/LifecycleTask.js]: nice I like that, will edit :)

File: tests/functional/lifecycle/LifecycleTask.js [hunk 5]
```diff
@@ -0,0 +1,511 @@
...
+                Key: key,
+                Tagging: this._scenario[scenarioNumber].tags[i],
+            }, done);
+        }, err => {
+            assert.ifError(err);
+            return cb();
+        });
+    }
+
+    emptyAndDeleteBucket(cb) {
```
Comment (jonathan-gramain) [inline tests/functional/lifecycle/LifecycleTask.js]: We have this implemented as `empty()` function in S3 repo at tests/functional/aws-node-sdk/lib/utility/bucket-util.js, ideally we would share one implementation across projects, for now it's fine.
